forcedotcom/salesforcedx-templates
# Revert to eslint 8 - for now

### What does this PR do?
Dependabot bumped `eslint` from 8 to 9. Version 9 is not supported by a number of repos that are not owned by the CLI team. It will take a coordinated effort to bring all of these up to date with eslint version 9. 

This reverts back to version 8 and also sets dependabot to ignore major version bumps. 

### Testing
- Pull this branch
- Run `yarn compile`
- Run `yarn link`
- `cd` to `plugin-templates`
- Run `yarn link "@salesforce/templates"`
- Run `yarn compile`
- Run `bin/run.js project generate --output-dir tmp --name foo`
- Run `cd tmp/foo`
- Run `npm install`
- It should install

### What issues does this PR fix or reference?
This work: 
[@W-17005525@](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-17005525)
https://github.com/forcedotcom/cli/issues/3065

Future work: 
[@W-16958727@](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-16958727)

Related:
https://github.com/forcedotcom/cli/issues/3056
https://github.com/forcedotcom/cli/issues/3042
https://github.com/forcedotcom/cli/issues/2986
